code/datums/status_effects/debuffs.dm

```diff
@@ -67,8 +67,19 @@
 	return ..()
 
 /datum/status_effect/incapacitating/sleeping/tick()
-	if(owner.getStaminaLoss())
-		owner.adjustStaminaLoss(-0.5) //reduce stamina loss by 0.5 per tick, 10 per 2 seconds
+	if(owner.maxHealth)
+		var/health_ratio = owner.health / owner.maxHealth
+		if(health_ratio > 0.8)
+			var/healing = -0.2
+			if((locate(/obj/structure/bed) in owner.loc))
+				healing -= 0.3
+			else
+				if((locate(/obj/structure/table) in owner.loc))
+					healing -= 0.1
+			owner.adjustBruteLoss(healing)
+			owner.adjustFireLoss(healing)
+			owner.adjustToxLoss(healing, TRUE, TRUE)
```

Comment:
I'd like toxloss healed slower than the B's since 1 point of damage is worth more than 1 point of brute/burn.

Reply:
after r2m

Reply:
so healing * 0.5 or something?

Reply:
that's fine

Reply:
boom